src/main.ts

```diff
@@ -94,9 +96,8 @@ const main = async () => {
     }
   }
 
-  console.log(`Saving to AirTable.`);
-
   await saveToAirTable(airtableToken, volunteers);
+  await uploadSubscribers(ecomailToken, volunteers);
```

Comment:
Pokud tahle funkce selže (tj. vrátí status code >= 400, viz https://www.npmjs.com/package/axios#handling-errors), tak se o tom nijak nedozvíme. `await` funkce jdou normálně dát do try/catch bloku a výjimku odchytit a zpracovat.

Reply:
Takže třeba takhle je to cajk?

```typescript
async function uploadSubscribers(
  apiKey: string,
  volunteers: Volunteer[]
): Promise<boolean> {
  try {
    const response = await axios({
      method: "post",
      url: "http://api2.ecomailapp.cz/lists/2/subscribe-bulk",
      data: buildSubscriberData(volunteers),
      headers: {
        key: apiKey
      }
    });
    return response.status < 400;
  } catch (e) {
    console.error(e);
    return false;
  }
}
```

Reply:
Nakonec jsem došel k tomuto:

```typescript
async function uploadSubscribers(
  apiKey: string,
  volunteers: Volunteer[]
): Promise<any> {
  return axios({…});
}
```

Tohle vrátí resolved promise pro stavové kódy >= 200 && < 300 a rejected promise pro cokoliv dalšího, takže si s tím může volající naložit, jak chce. (Momentálně nemám ve volajícím kódu `try/catch`, přidal bych.)

PS. To `any` v `Promise<any>` beru jako „opaque value“, slušňácké řešení by bylo přepsat na `Promise<void>` („když nedostaneš výjimku, dopadlo to dobře“) a mapovat výstup z axiosu `.then(_ => {})`, ale to už mně přijde zbytečný.